actors/builtin/market/market_test.go

```diff
@@ -1359,6 +1389,31 @@ func TestCronTickDealExpiry(t *testing.T) {
 	endEpoch := abi.ChainEpoch(300)
 	sectorExpiry := abi.ChainEpoch(400)
 
+	t.Run("deal expiry -> deal is correctly processed twice in the same crontick", func(t *testing.T) {
+		end := abi.ChainEpoch(151) // 50 + 101
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		dealId := actor.publishAndActivateDeal(rt, client, mAddrs, startEpoch, end, 0, sectorExpiry)
+		d := actor.getDealProposal(rt, dealId)
+
+		// move the current epoch to startEpoch so next cron epoch will be start + 100 = 150
+		current := startEpoch
+		rt.SetEpoch(current)
+		pay, slashed := actor.cronTickAndAssertBalances(rt, client, provider, current, dealId)
+		require.EqualValues(t, big.Zero(), pay)
+		require.EqualValues(t, big.Zero(), slashed)
+
+		// move the epoch to 155(anything greater than 150), so deal is first processed at 150 & then at 151 which is it's end epoch
+		// total payment = (end - start) = 151 - 50 = 101
+		current = 155
+		rt.SetEpoch(current)
+		pay, slashed = actor.cronTickAndAssertBalances(rt, client, provider, current, dealId)
```

Comment:
Is there a way to ensure this processed multiple epochs? (or maybe this exists in some other test?)

Reply:
Maybe this should be added to `cronTickAndAssertBalances` - assert that multiple epochs were successfully processed.

Reply:
@wadeAlexC I am not sure inspecting internal states like that is a good idea for tests. We have tests that ensure that deals in different epochs are process in a `CronTick`. Coupled with this,  it is proof enough.